File: foxy/parent/src/command/vanilla/actions/DanceExecutor.ts
import { bot } from "../../../FoxyLauncher";
import { createEmbed } from "../../../utils/discord/Embed";
import { ExecutorParams } from "../../structures/CommandExecutor";

export default class DanceExecutor {
    async execute({ context, endCommand, t }: ExecutorParams) {
        const embed = createEmbed({});
        const danceGif = await bot.rest.foxy.getImage("roleplay", "dance");

        embed.title = t('commands:dance.dancing', { author: await bot.rest.foxy.getUserDisplayName((await context.getAuthor()).id) }),
            embed.image = {
                url: danceGif.url
            }

        context.reply({
            embeds: [embed],
        })
        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/actions/DanceExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.actions

import net.cakeyfox.foxy.interactions.FoxyInteractionContext
import net.cakeyfox.foxy.interactions.commands.FoxySlashCommandExecutor

class DanceExecutor : FoxySlashCommandExecutor() {
    override suspend fun execute(context: FoxyInteractionContext) {
        context.defer()

        val response = context.foxy.utils.getActionImage("dance")

        context.reply {
            embed {
                description = context.locale["dance.description", context.user.asMention]
                image = response
            }
        }
    }
}